src/commands/PlayCommand.ts

```diff
@@ -31,73 +31,122 @@ export class PlayCommand extends BaseCommand {
         }
         const searchString = args.join(" ");
         const url = searchString.replace(/<(.+)>/g, "$1");
+        const urlData = this.checkUrlData(url);
 
         if (message.guild?.queue !== null && voiceChannel.id !== message.guild?.queue.voiceChannel?.id) {
             return message.channel.send(
                 createEmbed("warn", `The music player is already playing to **${message.guild?.queue.voiceChannel?.name}** voice channel`)
             );
         }
 
-        if (/^https?:\/\/(www\.youtube\.com|youtube.com)\/playlist(.*)$/.exec(url)) {
-            try {
-                const id = new URL(url).searchParams.get("list")!;
-                const playlist = await this.client.youtube.getPlaylist(id);
-                const videos = await playlist.getVideos();
-                let skippedVideos = 0;
-                const addingPlaylistVideoMessage = await message.channel.send(
-                    createEmbed("info", `Adding all videos in **[${playlist.title}](${playlist.url})** playlist, please wait...`)
-                        .setThumbnail(playlist.thumbnailURL)
-                );
-                for (const video of Object.values(videos)) {
-                    if (video.isPrivate) {
-                        skippedVideos++;
-                        continue;
-                    } else {
-                        const video2 = await this.client.youtube.getVideo(video.id);
-                        await this.handleVideo(video2, message, voiceChannel, true);
+
+        if (urlData.source === "youtube") {
+            if (urlData.type === "playlist") {
+                try {
+                    const id = new URL(url).searchParams.get("list")!;
+                    const playlist = await this.client.youtube.getPlaylist(id);
+                    const videos = await playlist.getVideos();
+                    let skippedVideos = 0;
+                    const addingPlaylistVideoMessage = await message.channel.send(
+                        createEmbed("info", `Adding all videos in **[${playlist.title}](${playlist.url})** playlist please wait...`)
```

Comment:
Why do you delete this, comma is needed in this statement
```suggestion
                        createEmbed("info", `Adding all videos in **[${playlist.title}](${playlist.url})** playlist, please wait...`)
```